shared/utility/math/coordinates.hpp

```diff
@@ -51,19 +51,23 @@ namespace utility::math::coordinates {
 
     template <typename T, typename U = typename T::Scalar>
     inline Eigen::Matrix<U, 3, 1> cartesianToSpherical(const Eigen::MatrixBase<T>& cartesianCoordinates) {
```

Comment:
I suppose @KipHamiltons wants these to be `[[nodiscard]]`s as well
```suggestion
    [[nodiscard]] inline Eigen::Matrix<U, 3, 1> cartesianToSpherical(const Eigen::MatrixBase<T>& cartesianCoordinates) {
```